pkg/vpcmodel/explainabilityConnectivity.go

```diff
@@ -135,79 +133,77 @@ func (c *VPCConfig) computeExplainRules(srcNodes, dstNodes []Node,
-// considering filtersExternal (which was computed based on the RoutingResource) and removing filters
-// not relevant for the Router e.g. nacl not relevant fip
-// 4. ingressEnabled and egressEnabled: whether traffic is enabled via ingress, egress
-
-func (details *rulesAndConnDetails) computeAdditionalDetails(c *VPCConfig) error {
+// 2. The external filters relevant to the <src, dst> given the routingResource
+// 3. The internal filters relevant to the <src, dst>
+// 4. The actual relevant filter, depending on whether src xor dst is external
+func (details *rulesAndConnDetails) computeFilters(c *VPCConfig) error {
 	for _, singleSrcDstDetails := range *details {
-		src := singleSrcDstDetails.src
-		dst := singleSrcDstDetails.dst
 		// RoutingResources are computed by the parser for []Nodes of the VPC,
 		// finds the relevant nodes for the query's src and dst;
 		// if for src or dst no containing node was found, there is no router
-		var routingResource RoutingResource
-		var filtersForExternal map[string]bool
-		var err error
-
-		routingResource, _, err = c.getRoutingResource(src, dst)
-		if err != nil {
-			return err
-		}
-		if routingResource != nil {
-			filtersForExternal = routingResource.AppliedFiltersKinds() // relevant filtersExternal
+		src := singleSrcDstDetails.src
+		dst := singleSrcDstDetails.dst
+		if src.IsInternal() && dst.IsInternal() { // internal
+			// security group applied always between two vsis
+			singleSrcDstDetails.filtersRelevant = map[string]bool{SecurityGroupLayer: true}
+			// nacl applied between two vsis if not same subnet
+			singleSrcDstDetails.filtersRelevant[NaclLayer] =
+				src.(InternalNodeIntf).Subnet().UID() != dst.(InternalNodeIntf).Subnet().UID()
+		} else { // external
+			routingResource, _, err := c.getRoutingResource(src, dst)
+			if err != nil {
+				return err
+			}
+			if routingResource == nil {
+				continue // no router: no connections, filtersLayers non defined
+			}
+			singleSrcDstDetails.router = routingResource
+			singleSrcDstDetails.filtersRelevant = routingResource.AppliedFiltersKinds() // relevant filtersExternal
 		}
+	}
+	return nil
+}
 
-		singleSrcDstDetails.router = routingResource
-		singleSrcDstDetails.filtersExternal = filtersForExternal
-		isInternal := singleSrcDstDetails.src.IsInternal() && singleSrcDstDetails.dst.IsInternal()
+// computeActualRules computes, after potentialRules and filters were computed, for each  <src, dst> :
+// 1. from the potentialRules the actualRules (per ingress, egress) that actually enable traffic,
+// considering filtersExternal (which was computed based on the RoutingResource) and removing filters
+// not relevant for the Router e.g. nacl not relevant fip
+// and considering filterInternal - removing nacl when both vsis are of the same subnets
+// 2. ingressEnabled and egressEnabled: whether traffic is enabled via ingress, egress
+func (details *rulesAndConnDetails) computeActualRules() {
```

Comment:
now there are two `computeActualRules` functions?

Reply:
no, there is one w.r.t. `srcDstDetails.filtersRelevant` which is internal or external, depending on src and dst

Reply:
there is both function `func computeActualRules(rulesLayers *rulesInLayers, filters map[string]bool) ` and function `func (details *rulesAndConnDetails) computeActualRules() {`
consider renaming one of them to avoid confusion.

Reply:
renamed and added documentation to make the difference clearer